src/runtime/cuda/nccl_api.cu

```diff
@@ -601,6 +612,24 @@ void NCCLCommunicator::AllToAllV(
     }
   }
   NCCL_CALL(ncclGroupEnd());
+  #else
+  CHECK_EQ(send_prefix[1]-send_prefix[0], recv_prefix[1]-recv_prefix[0]) <<
+      "Send message size must equal receive message size.";
```

Comment:
Does this due to the limitation of device->CopyDataFromTo?

Reply:
Yes, but I'm not sure I'd call it a limitation. Since our communicator is size 1 in the case, there is only one message, so it's size must be identical both from the buffer it's being copied from and the buffer it's being copied to.